### PR TITLE
feat(ls): provide OpenAPI 3.1.0 completion rules for Responses object

### PR DESCRIPTION
**Pull request by tim-lai**
### Description

* provide OpenAPI 3.1.0 completion rules for Responses object

### Motivation and Context

Ref: #2062 
Ref: #2061 (no update required)

### My PR contains...

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [x] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [ ] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by tim-lai** [comment]
@char0n I pretty matched the OAS3.0.x version, however, how to best test the response codes? Even without this PR, I'm not getting a suggestion for auto complete in OAS3.0.x definition for `'20...'` (string) or `20...` (number). With partial http code and without closing `:`, I do get the error that `responses` should contain an object.

File: packages/apidom-ls/src/config/openapi/responses/completion.ts [hunk 1]
```diff
@@ -23,211 +23,266 @@ const httpCode3_0CompletionRule = {
...
+  documentation: {
+    kind: 'markdown',
+    value:
+      '[Response Object](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#responseObject) \\| [Reference Object](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#referenceObject)\n\\\n\\\nThe documentation of responses other than the ones declared for specific HTTP response codes. Use this field to cover undeclared responses.',
+  },
+  targetSpecs: [{ namespace: 'openapi', version: '3.1.0' }],
+};
+
 const completion: ApidomCompletionItem[] = [
   {
```
Comment (char0n) [inline packages/apidom-ls/src/config/openapi/responses/completion.ts:41]: You cannot do this, you need to duplicate every rules. After this change things will work as expected

```s
  {
    label: 'default',
    insertText: 'default',
    ...httpCode3_0CompletionRule,
  } as ApidomCompletionItem,
  {
    label: 'default',
    insertText: 'default',
    ...httpCode3_1CompletionRule,
  } as ApidomCompletionItem,

**Review by char0n — CHANGES_REQUESTED** [review]
I've suggested change that will make the things work for you again.

**Comment by tim-lai** [comment]
> I've suggested change that will make the things work for you again.

Changes made. I see completion working when under `paths/path-item/operation/responses`, but not `components/responses`.

**Comment by char0n** [comment]
> > I've suggested change that will make the things work for you again.
> 
> Changes made. I see completion working when under `paths/path-item/operation/responses`, but not `components/responses`.

Here is the type of `Operation Object`.`responses` field

image

Here is the type of `Components Object`.`responses` field

image

Notice the difference. To say it simply - the `Components Object`.`responses` field if of type of generic map/object instead of `Responses Object`. So what you're seeing is correct and this PR can be merged.

**Review by char0n — APPROVED** [review]
This is now LEGIT.
